Userland/Applications/PixelPaint/FilterTreeModel.cpp

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2021-2022, Tobias Christiansen <tobyase@serenityos.org>
+ * Copyright (c) 2021, Mustafa Quraish <mustafa@serenityos.org>
+ * Copyright (c) 2022, Linus Groh <linusg@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "FilterTreeModel.h"
+#include "Filters/Bloom.h"
+#include "Filters/BoxBlur3.h"
+#include "Filters/BoxBlur5.h"
+#include "Filters/FastBoxBlur.h"
+#include "Filters/GaussBlur3.h"
+#include "Filters/GaussBlur5.h"
+#include "Filters/Grayscale.h"
+#include "Filters/Invert.h"
+#include "Filters/LaplaceCardinal.h"
+#include "Filters/LaplaceDiagonal.h"
+#include "Filters/Sepia.h"
+#include "Filters/Sharpen.h"
+#include <LibGUI/FileIconProvider.h>
+
+namespace PixelPaint {
+
+ErrorOr<NonnullRefPtr<GUI::TreeViewModel>> create_filter_tree_model(ImageEditor* editor)
+{
+    auto directory_icon = GUI::FileIconProvider::directory_icon();
+    auto filter_icon = GUI::Icon(TRY(Gfx::Bitmap::try_load_from_file("/res/icons/pixelpaint/filter.png")));
+
+    auto filter_tree_model = GUI::TreeViewModel::create();
+
+    auto add_filter_node = [&]<typename FilterType>(GUI::TreeViewModel::Node& node) {
+        auto filter = adopt_own(*new FilterType(editor));
+        (void)node.add_node<FilterNode>(filter->filter_name(), filter_icon, move(filter));
```

Comment:
This error can just be returned, and the call wrapped in `TRY`

Reply:
That's not an error, just NNRP being `[[nodiscard]]`